### PR TITLE
Pages redesign 

### PR DESCRIPTION
**Pull request by alexnguyenn**
## Changes 
* Switching from bootstrap to Chakra UI. Overhaul the design and added options for dark and light mode.
* Splitting into multiple pages using React Routers
* Added new Home page. 
* Added About Us (available at [`/about-us`](https://latticesurgery-com-web-ui-pr78-preview.surge.sh/about-us)). 
* Added Overview page (available at [`/overview`](https://latticesurgery-com-web-ui-pr78-preview.surge.sh/overview)).
* Compiler is moved to [`/online-compiler`](https://latticesurgery-com-web-ui-pr78-preview.surge.sh/online-compiler)), new design for the uploader, eliminate duplicating logic between `SelectExampleCircuit` and `UploadCircuit` (by combining them). 
* Updated favicons based on the new logo.

### New dependencies added
* [ChakraUI](https://chakra-ui.com/)
* [React Router](https://reactrouter.com/)
* [React Icons](https://react-icons.github.io/react-icons/)
* [react-dropzone](https://github.com/react-dropzone/react-dropzone/)

### Dependencies to be removed
* react-css-spinners ([Chakra UI already has its own](https://chakra-ui.com/docs/feedback/spinner))
* Bootstrap (We are still using the bootstrap CDN for `LatticeView`, would have to update that later)
## Context 
Closes https://github.com/latticesurgery-com/web-ui/issues/2. Bootstrap is a good framework, however it was not designed with React in mind. I am suggesting that we switch over to a component-based UI framework ([Chakra](https://chakra-ui.com/)). 

~~I am also suggesting that we start adding more pages into the website (rather only having a main page like right now). LMK what you guys want to add.~~ Has been discussed.

### CODE REVIEW

**Comment by alexnguyenn** [comment]
pinging @Keelando and @gwwatkin for discussions. Also seems like the about page is not working with Surge deployment yet - I will take a look later. For now you would have to checkout the branch and see the new page yourself I guess.

File: src/pages/AboutUs.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    maxW={"320px"}
+                    w={"full"}
+                    boxShadow={"2xl"}
+                    rounded={"md"}
+                    p={"6"}
+                    textAlign={"center"}
+                >
+                    <Avatar src="https://avatars.githubusercontent.com/u/47836064" mb={4} />
+                    <Heading fontSize={"2xl"} fontFamily={"body"}>
+                        Keeland Watkins
```
Comment (Keelando) [inline src/pages/AboutUs.tsx:83]:
```suggestion
                        Keelan Watkins
```

**Comment by gwwatkin** [comment]
@alexnguyenn the about page looks awesome! can't wait to see the main page restyled.

There is an ongoing debate about the compilation view being on the same page on a separate page. I suspect that given that we want the visualizer to work for 100s of patches (#77) it might be best to put the lattice view into a new page. I think you should discuss this with @Keelando. However I still think the upload circuit/try example form should still be the first thing on the index page ... just that it will load a new page once the spinner is done.

As for the for the text of the main page I do think we need to make it better and more direct.

**Review by Keelando — COMMENTED** [review]
About-us page looks good. Perhaps a subtle color gradient instead of white background could be interesting. I am still in favor of keeping it one page. With react, you can turn on or off different sections of the page, so perhaps we could have it towards the bottom of the main page?

And we'll definitely need to have a short version of the about text on the main page. 2 to 3 sentences tops.

I am not going to deter you from developing a new UI theme with Chakra, but you may be on your own on learning it initially, as I still feel like I have my hands full with React/Typescript.

**Comment by gwwatkin** [comment]
@alexnguyenn What do you think about cards Aleandru, Kero and Steve as advisors below?

**Comment by gwwatkin** [comment]
@alexnguyenn I will get to improving the content on this page

**Comment by alexnguyenn** [comment]
@gwwatkin @Keelando I have fixed the issue with Surge, now you can go to see the new pages on the preview itself.

**Comment by gwwatkin** [comment]
I see, might be tricky, will try on dev first

**Comment by alexnguyenn** [comment]
I think I won't implement the file uploader on the home page right now, that will be a future pull request. My remaining focus is to redesign the file uploader and update the content to get this out of the way so it won't block developments.

**Review by gwwatkin — APPROVED** [review]
Superb changes. Good to merge after @alexnguyenn, @Keelando  and @isolatedinformation  add their bios (unless they prefer lorem ipsum :rofl: ).

Made a few content changes.

Also created the following to complement the functionality some other time, but we definitely want this in by today.
 - https://github.com/latticesurgery-com/web-ui/issues/91
 - https://github.com/latticesurgery-com/web-ui/issues/92
 - https://github.com/latticesurgery-com/web-ui/issues/93

**Pull request merged**